# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoadRequest.h
# Commit: b8ecf23519c9
# Full Hash: b8ecf23519c9ab5180d08f00c712674bfe968dac
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-21 21:39:46
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 85fd9fc9cffa -r b8ecf23519c9 js/loader/ModuleLoadRequest.h
--- a/js/loader/ModuleLoadRequest.h	Mon Feb 21 15:39:08 2022 +0000
+++ b/js/loader/ModuleLoadRequest.h	Mon Feb 21 15:39:09 2022 +0000
@@ -4,18 +4,18 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#ifndef mozilla_dom_ModuleLoadRequest_h
-#define mozilla_dom_ModuleLoadRequest_h
+#ifndef js_loader_ModuleLoadRequest_h
+#define js_loader_ModuleLoadRequest_h
 
 #include "ScriptLoadRequest.h"
+#include "ModuleLoaderBase.h"
 #include "mozilla/MozPromise.h"
 #include "js/RootingAPI.h"
 #include "js/Value.h"
 #include "nsURIHashKey.h"
 #include "nsTHashtable.h"
 
-namespace mozilla {
-namespace dom {
+namespace JS::loader {
 
 class ModuleScript;
 class ModuleLoaderBase;
@@ -43,10 +43,15 @@
   NS_DECL_ISUPPORTS_INHERITED
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS_INHERITED(ModuleLoadRequest,
                                                          ScriptLoadRequest)
+  using SRIMetadata = mozilla::dom::SRIMetadata;
+
+  template <typename T>
+  using MozPromiseHolder = mozilla::MozPromiseHolder<T>;
+  using GenericPromise = mozilla::GenericPromise;
 
   ModuleLoadRequest(nsIURI* aURI, ScriptFetchOptions* aFetchOptions,
                     const SRIMetadata& aIntegrity, nsIURI* aReferrer,
-                    ScriptLoadContext* aContext, bool aIsTopLevel,
+                    mozilla::dom::ScriptLoadContext* aContext, bool aIsTopLevel,
                     bool aIsDynamicImport, ModuleLoaderBase* aLoader,
                     VisitedURLSet* aVisitedSet, ModuleLoadRequest* aRootModule);
 
@@ -114,7 +119,6 @@
   JS::Heap<JSObject*> mDynamicPromise;
 };
 
-}  // namespace dom
-}  // namespace mozilla
+}  // namespace JS::loader
 
-#endif  // mozilla_dom_ModuleLoadRequest_h
+#endif  // js_loader_ModuleLoadRequest_h